libkbfs/keybase_service_base.go

```diff
@@ -446,6 +447,25 @@ func (k *KeybaseServiceBase) Notify(ctx context.Context, notification *keybase1.
 	return k.kbfsClient.FSEvent(ctx, *notification)
 }
 
+// NotifySyncStatus implements the KeybaseService interface for
+// KeybaseServiceBase.
+func (k *KeybaseServiceBase) NotifySyncStatus(ctx context.Context,
+	status *keybase1.FSPathSyncStatus) error {
+	// Reduce log spam by not repeating log lines for
+	// notifications with the same pathname.
+	//
+	// TODO: Only do this in debug mode.
+	func() {
+		k.lastNotificationFilenameLock.Lock()
+		defer k.lastNotificationFilenameLock.Unlock()
+		if status.Pathname != k.lastSyncNotificationPathname {
+			k.lastSyncNotificationPathname = status.Pathname
+			k.log.CDebugf(ctx, "Sending notification for %s", status.Pathname)
+		}
+	}()
+	return k.kbfsClient.FSSyncEvent(ctx, *status)
```

Comment:
No, if the pathname is the same I just want to skip the log message (we do the same thing in `Notify` above).  It's just to reduce log spam.

(Also note that in the current revision, since the path name is never set, there will be no log messages.)